pimd/pim_nht.c

```diff
@@ -499,16 +501,40 @@ int pim_ecmp_nexthop_search(struct pim_instance *pim,
 			}
 		}
 	}
+
+	/*
+	 * Look up all interfaces and neighbors,
+	 * store for later usage
+	 */
+	for (nh_node = pnc->nexthop, i = 0; nh_node;
+	     nh_node = nh_node->next, i++) {
+		ifps[i] = if_lookup_by_index(nh_node->ifindex, pim->vrf_id);
+		if (ifps[i]) {
+			nbrs[i] = pim_neighbor_find(ifps[i],
+						    nh_node->gate.ipv4);
+			if (nbrs[i] || pim_if_connected_to_source(ifps[i],
+								  src->u.prefix4))
+				num_nbrs++;
+		}
+	}
 	if (pim->ecmp_enable) {
+		uint32_t consider = pnc->nexthop_num;
+
+		if (neighbor_needed && num_nbrs < consider)
```

Comment:
Why are we trying to do here?

Reply:
It's simple.  Suppose I have 4 way ecmp on a rpf check, but only have 3 pim neighbors.  If that is so we only can consider 3 paths for rpf usage.  If we do not do this we can get into situations where the rpf check fails.

Reply:
I was thrown off by the &&